network.go

```diff
@@ -852,6 +852,10 @@ func (n *network) ipamAllocate() ([]func(), error) {
 				if ip = net.ParseIP(v); ip == nil {
 					return nil, types.BadRequestErrorf("non parsable secondary ip address %s (%s) passed for network %s", k, v, n.Name())
 				}
+				if !d.Pool.Contains(ip) {
```

Comment:
Your second option is doable as well, but then we have to request the big pool (to get the correct id) and then we can reserve the address. We'd then have to either store the big pool id or retrieve the config on network release.
For the config we need to persist it (currently we don't)

Seems a bit convoluted.
Not sure what best to do.